js-miniapp-sample/src/pages/user-details.js

```diff
@@ -235,6 +267,38 @@ function UserDetails(props: UserDetailsProps) {
       });
   }
 
+  function requestPoints() {
+    const permissionsList = [
+      {
+        name: CustomPermissionName.POINTS,
+        description:
+          'We would like to display your Points on your profile page.',
+      },
+    ];
+
+    props
+      .requestPermissions(permissionsList)
+      .then((permissions) =>
+        permissions
+          .filter(
+            (permission) => permission.status === CustomPermissionStatus.ALLOWED
+          )
+          .map((permission) => permission.name)
```

Comment:
Looks like this is the same code used above to check for allowed permissions. So maybe you could extract it into a `filterAllowedPermissions` function?